Feature/movement [In Progress]

kodomazer opened the pull request:
Slowly putting together the movement handling.

brianwzhao committed:
Starting Camera Position

Move the starting camea position up so it's in a good position even without a VR setup

brianwzhao committed:
Rotate Ships

Have the ships facing positive X

brianwzhao committed:
Unit Selection

Select units with Mouse Selector via Unit Manager.  Currently always additive selection.

Maybe we want to add ToggleSelection to the Selection Manager

brianwzhao committed:
Basic Movement

Things move
No checks are made for restrictions yet

Possibly want to change mouseSelector to something else.  What do we call non-VR?

SeijiEmery commented on Assets/Script/PlayerControls/MouseSelector.cs:
```diff
@@ -0,0 +1,108 @@
+        ////////////////////////////
+        Vector3 movement=new Vector3();
+        Vector3 aim = new Vector3();
+        if (Input.GetKey(KeyCode.W))
+        {
+            movement.z += 1;
+        }
+        if (Input.GetKey(KeyCode.A))
+        {
+            movement.x -= 1;
+        }
+        if (Input.GetKey(KeyCode.S))
+        {
+            movement.z -= 1;
+        }
+        if (Input.GetKey(KeyCode.D))
+        {
+            movement.x += 1;
+        }
+        if (Input.GetKey(KeyCode.Q))
+        {
+            aim.z -= 1;
+        }
+        if (Input.GetKey(KeyCode.E))
+        {
+            aim.z += 1;
+        }
+        if (Input.GetKey(KeyCode.LeftShift))
+        {
+            movement.y -= 1;
+        }
+        if (Input.GetKey(KeyCode.Tab))
+        {
+            movement.y += 1;
+        }
```
could use Input.GetAxis() but this is fine

SeijiEmery commented on Assets/Script/PlayerControls/MouseSelector.cs:
```diff
@@ -0,0 +1,108 @@
+        {
+            selectable.highlighted = false;
+        }
+    }
```
could refactor to SetHighlighted(GameObject, bool)

SeijiEmery commented on Assets/Script/KelplerUnits/UnitMovement.cs:
```diff
@@ -21,17 +22,96 @@ public class UnitMovement : MonoBehaviour {
+
+        aimLR.SetPosition(0, transform.position);
+        aimLR.SetPosition(1, Aim);
+    }
+
+    private void FixedUpdate()
+    {
+        float deltaTime = Time.fixedDeltaTime;
+        Vector3 deltaPos = Destination - transform.position;
+        if (deltaPos.sqrMagnitude > moveSpeed * moveSpeed * deltaTime * deltaTime)
+        {
+            deltaPos = deltaPos.normalized * moveSpeed * deltaTime;
+        }
+        transform.position += deltaPos;
+        Aim += deltaPos;
+
+    }
+
     // sets target position that unit should move to
     // should ignore y component if unit cannot move vertically
     // TODO: implement this
-    public void SetMovementTarget(Vector3 targetPos) { }
+    public void SetMovementTarget(Vector3 targetPos)
+    {
+        Destination = targetPos;
+    }
 
     // sets firing target position for this unit
     // TODO: implement this
-    public void SetFiringTarget(Vector3 targetPos) { }
+    public void SetFiringTarget(Vector3 targetPos)
+    {
+        Aim = targetPos;
+    }
 
     // sets movement direction for direct (WASD) movement
     // moveDir() should be normalized before using
     // TODO: implement this
-    public void SetMovementDirection(Vector3 moveDir) { }
+    public void MoveInDirection(Vector3 moveDir)
```
I think I explicitly named this 'SetMovement' b/c you're telling the unit to move in a direction (and if you don't send a command to stop moving it won't). This was pretty clunky in general though, so yeah renaming this is fine

SeijiEmery reviewed (commented):
Looks good so far

FreemanMakesGames committed:
SelectableUnit has functions to handle selecting and hovering now, replacing the handling in Update.

SeijiEmery committed:
Merge pull request #5 from SeijiEmery/minor-optimization-on-highlighting

Minor optimization on highlighting

brianwzhao committed:
Remove Aim Line

brianwzhao committed:
Restrict Vertical Movement

Based on flags set on the object

brianwzhao committed:
Clamp Movement

Added movement restrictions to the Selection Manager.  Preliminary bounds set as well.

SeijiEmery committed:
Merge branch 'master' of https://github.com/SeijiEmery/KelplerEffectPrototype into feature/movement
